docs/javascript/imports.md

```diff
@@ -1,118 +1,194 @@
+By default, npm imports resolve to the latest version of the given package. Imported versions are resolved on build, or during preview after you clear your [npm cache](#self-hosting-of-npm-imports) and restart the server. To load an earlier or specific version of a package, add a [semver range](https://docs.npmjs.com/about-semantic-versioning). For example, to load major version 1 of canvas-confetti:
 
 ```js run=false
-import confetti from "https://cdn.jsdelivr.net/npm/canvas-confetti/+esm";
+import confetti from "npm:canvas-confetti@1";
 ```
 
-<div class="note">We plan on supporting importing from <code>node_modules</code> using bare module specifiers (<i>e.g.</i>, <code>import confetti from "canvas-confetti"</code>). If you’d like this feature, please upvote <a href="https://github.com/observablehq/framework/issues/360">#360</a>.</div>
-
-To load a specific version of a library, add a semver range:
+If the import path is not specified, it defaults to the default entry point is determined by the `package.json`; see [jsDelivr’s GitHub](https://github.com/jsdelivr/jsdelivr/issues/18263) for details. To load a different entry point, specify the desired path. For example, to load mime’s `lite` entry point:
 
 ```js run=false
-import confetti from "npm:canvas-confetti@1";
+import mime from "npm:mime/lite";
 ```
 
-To load a specific entry point, add a slash and the desired path:
+Similarly, to load the file `dist/confetti.module.mjs` from canvas-confetti:
 
 ```js run=false
-import confetti from "npm:canvas-confetti@1/dist/confetti.module.mjs";
+import confetti from "npm:canvas-confetti/dist/confetti.module.mjs";
 ```
 
-If you’re having difficulty getting an import working, it may help to browse the package and see what files are available as well as what’s exported in the `package.json`. You can browse the contents of a published module via jsDelivr; for example, for `canvas-confetti` see <https://cdn.jsdelivr.net/npm/canvas-confetti/>.
+<div class="tip">If you’re having difficulty importing, it may help to browse the package and see what files are available, and what’s exported in the <code>package.json</code>. You can browse the contents of a published module via jsDelivr; for example, for canvas-confetti see <a href="https://cdn.jsdelivr.net/npm/canvas-confetti/">https://cdn.jsdelivr.net/npm/canvas-confetti/</a>.</div>
 
 ### Self-hosting of npm imports
 
-Framework automatically downloads `npm:` imports from jsDelivr during preview and build. This improves performance and security of your built site by removing external code dependencies. It also improves performance during local preview by only downloading libraries once.
+Framework downloads `npm:` imports from jsDelivr during preview and build. This improves performance and security of your built site by removing runtime dependencies on external sites. It also improves performance during local preview by only downloading libraries once.
```

Comment:
```suggestion
Framework downloads `npm:` imports from jsDelivr during preview and build. This improves performance and security of your built site and of the local preview by removing runtime dependencies on external sites.
```

Reply:
It doesn’t remove the dependency from preview, though, since you still have the dependency the first time you import the library (before it is cached). Also I think “runtime dependency” only makes sense in the context of build (where “runtime” refers to running the built site); during preview, everything is runtime.

Reply:
Okay, I removed this line, and instead added “An imported module is downloaded from jsDelivr only if it is not already in the cache” to the next paragraph, which I think is better because it explains a little more the nature of our caching: it never expires, and versions are resolved against what’s in the cache. Although I don’t say this so explicitly… and I don’t want to explain “satisfying” semantic versions _etc._ 😓